### PR TITLE
[NI-652,NI-689] 301 Redirects for subdomains and Geo

### PR DESCRIPTION
**Pull request by perangel**
This PR is for handling 301 redirects for sunset subdomains and also performing 301 redirects for international properties (eg. cn.udacity.com)

### CODE REVIEW

File: roles/haproxy/files/haproxy.tmpl [hunk 1]
```diff
@@ -101,9 +101,18 @@ frontend logger
...
+    # Check for 301 redirect
+    acl has_redirect hdr(host),map(/etc/haproxy/301-redirects.lst) -m found
+    http-request redirect prefix https://%[hdr(host),map(/etc/haproxy/301-redirects.lst)] code 301 if has_redirect
+
     # Inject X-Berlioz-Country header with two-letter country code (see NI-461)
     http-request set-header X-Berlioz-Country %[req.hdr(X-Forwarded-For,-1),map_ip(/etc/haproxy/geolocation.txt)]
 
+    # Redirect Geo-specific domains 
+    acl is_www hdr(host) -i www.{{env "HAPROXY_DOMAIN"}}
+    acl has_geo_domain hdr(X-Berlioz-Country),map(/etc/haproxy/301-geo-redirects.lst) -m found
```
Comment (earldouglas) [inline roles/haproxy/files/haproxy.tmpl]: Any way to limit this to a set of beta testers?  E.g.:

* Extra ACL that looks for a certain set of IP addresses that we can test from in China
* Extra ACL that looks for a certain secret header that testers manually inject

**Comment by earldouglas** [comment]
\o/

File: roles/haproxy/files/haproxy.tmpl (hunk 1 above)
Comment (perangel) [inline roles/haproxy/files/haproxy.tmpl]: Yes we could certainly do that. We could also just setup a dummy domain as the `is_www` condition and use that to test (eg. `cn-test.udacity.com`)

**Comment by pmbauer** [comment]
Lgtm

**Comment by earldouglas** [comment]
This is great.  How can we test it before it needs to go live?

**Comment by perangel** [comment]
I'm setting up a small instance in china for us and I'm gonna run VPN on it

**Comment by perangel** [comment]
We can also have folks in China

**Comment by perangel** [comment]
This is now just 301s for subdomains. Geo stuff was moved to Jakku

**Pull request merged**